zerver/lib/actions.py

```diff
@@ -4660,11 +4660,17 @@ def subscriber_info(user_id: int) -> Dict[str, Any]:
     send_event(user_profile.realm, event, users_to_be_notified)
     return len(changed_messages)
 
-
 def do_delete_messages(realm: Realm, messages: Iterable[Message]) -> None:
-    message_ids = []
+    message_ids = [message.id for message in messages]
+    usermessages = UserMessage.objects.filter(message_id__in=message_ids)
+    message_id_to_notifiable_users = {}  # type: Dict[int, List[Dict[str, int]]]
+    for um in usermessages:
+        if um.message_id not in message_id_to_notifiable_users:
+            message_id_to_notifiable_users[um.message_id] = []
+        message_id_to_notifiable_users[um.message_id].append({"id": um.user_profile_id})
```

Comment:
This can be a new 3rd commit, but I think it can just be a list (or maybe even Set) of IDs, rather than a list of these little dicts?  We'd also need to delete the `process_userdata_event` logic for delete events -- it seems like someone was confused and cargo-culted this dictionary arrangement off of the edit message code path unnecessarily.

Reply:
Ah, with ``process_userdata_event`` removed and the default ``process_event`` now being applied it works with just a list of ``id``s instead of those dicts. Is that the correct change? I pushed a commit.

Reply:
Tested manually, and deleting message correctly makes them disappear in another browser logged in as another user, so seems correct unless I'm missing some subtlety.